extensions/authentication-passport/src/strategy-adapter.ts

```diff
@@ -3,27 +3,36 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import {AuthenticationStrategy} from '@loopback/authentication';
-import {UserProfile} from '@loopback/security';
+import {
+  AuthenticationStrategy,
+  UserToUserProfileConverterFn,
+} from '@loopback/authentication';
 import {HttpErrors, Request} from '@loopback/rest';
+import {UserProfile} from '@loopback/security';
 import {Strategy} from 'passport';
 
 const passportRequestMixin = require('passport/lib/http/request');
-
 /**
  * Adapter class to invoke passport-strategy
  *   1. provides express dependencies to the passport strategies
  *   2. provides shimming of requests for passport authentication
- *   3. provides lifecycle similar to express to the passport-strategy
+ *   3. provides life-cycle similar to express to the passport-strategy
  *   4. provides state methods to the strategy instance
  * see: https://github.com/jaredhanson/passport
  */
-export class StrategyAdapter implements AuthenticationStrategy {
+export class StrategyAdapter<U> implements AuthenticationStrategy {
   /**
    * @param strategy instance of a class which implements a passport-strategy;
    * @description http://passportjs.org/
    */
-  constructor(private readonly strategy: Strategy, readonly name: string) {}
+  constructor(
+    private readonly strategy: Strategy,
+    readonly name: string,
+    // The default converter returns an user as user profile
+    private userConverter: UserToUserProfileConverterFn<U> = (u: unknown) => {
```

Comment:
This should be optional.

Reply:
@raymondfeng a default converter is provided (return the `user` as it is) therefore the converter here is not marked as optional :)

See discussion in https://github.com/strongloop/loopback-next/pull/3771#discussion_r326611630